Stack/Opc.Ua.Core/Stack/Tcp/TcpServerChannel.cs

```diff
@@ -566,13 +569,28 @@ private bool ProcessOpenSecureChannelRequest(
 
             try
             {
+                m_oscRequestSignature = null;
+                byte[] signature;
+
                 messageBody = ReadAsymmetricMessage(
                     messageChunk,
                     ServerCertificate,
                     out channelId,
                     out clientCertificate,
                     out requestId,
-                    out sequenceNumber);
+                    out sequenceNumber,
+                    m_oscRequestSignature,
+                    out signature);
+
+                // don't keep signature if secure channel enhancements are not used.
+                m_oscRequestSignature = (SecurityPolicy.SecureChannelEnhancements) ? signature : null;
+
+                CryptoTrace.Start(ConsoleColor.Magenta, $"ProcessOpenSecureChannelRequest ({(State != TcpChannelState.Opening ? "RENEW" : "OPEN")})");
```

Comment:
This should use logging infra, e.g. Logger.Trace.  If it is for debugging, it should be behind #if DEBUG.